concent_api/middleman/middleman_server.py

```diff
@@ -0,0 +1,83 @@
+import asyncio
+
+import socket
+import sys
+from logging import getLogger
+
+from middleman.decorators import safe_execution
+
+PORT = 8888
+
+
+logger = getLogger(__name__)
+
+
+class MiddleMan(object):
+    def __init__(self, host=None, port=None):
+        self.host = host or socket.gethostname()
+        self.port = port or PORT
+        self.loop = None
+        self.counter = 0
+        self._connection_map = {}
+        self._is_server_running = False
+
+    def start_server_for_concent(self):
+        if self._is_server_running:
+            raise RuntimeError("Server for Concent has been already started")
+        self._is_server_running = True
+        logger.info("Starting MiddleMan")
+        self.loop = asyncio.get_event_loop()
+
+        # start MiddleMan server
+        try:
+            server_coroutine = asyncio.start_server(self.handle_user_request, self.host, self.port, loop=self.loop)
+            server = self.loop.run_until_complete(server_coroutine)
+            # Serve requests until Ctrl+C is pressed
+            logger.info('MiddleMan is serving on {}'.format(server.sockets[0].getsockname()))
+            try:
+                self.loop.run_forever()
+            except KeyboardInterrupt:
+                pass
+            # Close the server
+            server.close()
+            self.loop.run_until_complete(server.wait_closed())
+            self.loop.close()
+        except Exception as exception:  # pylint: disable=broad-except
+            logger.error(f"Exception occurred while starting MiddleMan server for Concent: {str(exception)}")
+            sys.exit(1)
+
+    @safe_execution(logger)
+    async def handle_user_request(self, reader, writer):
+        data = await reader.readuntil()
+        addr = writer.get_extra_info('peername')
+        print("Received %r from %r" % (data, addr))
+        connection_id = self.counter
+        self.counter += 1
+        self._connection_map[connection_id] = writer
+        await self.respond_to_user(data, connection_id)
+
+    @safe_execution(logger)
+    async def respond_to_user(self, data: bytes, connection_id: int):
+        writer = self._connection_map.get(connection_id)
+        if writer is not None:
+            writer.write(data)
+            await writer.drain()
+            writer.close()
+            del self._connection_map[connection_id]
```

Comment:
Should the connection be instantly deleted ?

Reply:
1. This map mechanism is needed (see @cameel 's blueprint: https://github.com/golemfactory/concent/issues/597)
2. For Concent, each request is handeled via sparate connection (there is no "hard wire" for Concent clients)

Reply:
I'd prefer if the one who added it was also responsible for removing it. But it depends on how we organize the handlers. That's one of the reasons why I think the map should wait for the pull request that adds message routing. It's hard to evaluate it properly here without context.